File: REST-Server/src/main/java/pl/antonic/server/methods/ImageProcessingMethod.java
package pl.antonic.server.methods;

import java.util.List;

public class ImageProcessingMethod {

    private Methods id;
    private List<String> arguments;

    public Methods getId() {
        return id;
    }

    public void setId(String id) {
        this.id = Methods.valueOf(id);
    }

    public List<String> getArguments() {
        return arguments;
    }

    public void setArguments(List<String> arguments) {
        this.arguments = arguments;
    }
}

File: AndroidCV/app/src/main/java/pl/antonic/androidcv/methods/ImageProcessingMethod.kt
package pl.antonic.androidcv.methods

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class ImageProcessingMethod : Serializable {

    @SerializedName("id")
    @Expose
    var id: Methods

    @SerializedName("arguments")
    @Expose
    var arguments: List<String>

    constructor(id: Methods, arguments: List<String>) {
        this.id = id
        this.arguments = arguments
    }
}